pkg/sources/adapter/webhooksource/webhook.go

```diff
@@ -158,3 +197,38 @@ func healthCheckHandler(w http.ResponseWriter, _ *http.Request) {
 	w.Header().Set("Content-Type", "text/plain; charset=utf-8")
 	w.WriteHeader(http.StatusOK)
 }
+
+func sanitizeCloudEventAttributeName(name string) string {
+	// only lowercase accepted
+	name = strings.ToLower(name)
+
+	// strip non valid characters
+	needsStripping := false
+	for i := range name {
+		if !((name[i] >= 'a' && name[i] <= 'z') || (name[i] >= '0' && name[i] <= '9')) {
+			needsStripping = true
+			break
+		}
+	}
+
+	if needsStripping {
+		stripped := []byte{}
+		for i := range name {
+			if (name[i] >= 'a' && name[i] <= 'z') || (name[i] >= '0' && name[i] <= '9') {
+				stripped = append(stripped, name[i])
+			}
+		}
+		name = string(stripped)
+	}
+
+	// truncate if longer than 20 characters
+	if len(name) > 20 {
+		name = name[:20]
+	}
+
+	// data is a reserved element at CloudEvents
+	if name == "data" || name == "path" || name == "method" || name == "host" {
+		return "data0"
+	}
```

Comment:
So if one of the headers or query parameters will be named `data`/`path`/`method`/`name`/`host` it will be set as `data0` CE attribute? And if there are several of them (these reserved names are pretty common, IMO), they will override each other? Is there a way to use some prefix to avoid overriding?

Reply:
You mean if a user sends something like?

```
http://myhost/my/path?data=0&data=1
```

In that case (and the same case for headers) we receive a single element named "data" that will be translated to "data0" containing an array of values. That array will be iterated here and it will be appended a new integer to each element

https://github.com/triggermesh/triggermesh/blob/49f55ad70898f626890eb0ca45ee2dcb9e5676ba/pkg/sources/adapter/webhooksource/webhook.go#L168-L173

There is a chance of overlapping that has nothing to do with this function if querystring and headers contain the same variable names (very unlikely if you ask me), and that is something to clarify at user docs, I did not find a way to solve that case.

Reply:
https://github.com/triggermesh/triggermesh/pull/1028/commits/f789144ef77fcc583002f271e773906f89fbfc51 @tzununbekov ^^